### PR TITLE
Modular API, Steamboat integrated into DDBot, Optimizations, Promise Rejection Handling, Fix Play Command

### PR DESCRIPTION
**Pull request by EricRabil**

### CODE REVIEW

File: Source/Commands/Administration/stop.js [hunk 1]
```diff
@@ -5,6 +5,6 @@ module.exports = new Command("stop", "Stop the bot", null, [],
         response.reply("", response.embedFactory.createSuccessEmbed("Stopping", message.member).setDescription("The bot will now stop.")).then(msg => {
           client.destroy();
           process.exit();
-        });
+        }).catch(e => client.log(e, true));
```
Comment (AppleBetas) [inline Source/Commands/Administration/stop.js]: GOOD FUCKING IDEA

Comment (AppleBetas) [inline Source/Commands/Administration/stop.js:8]: GOOD FUCKING IDEA

File: Source/Managers/Modules.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const fs = require("fs-extra-promise");
```
Comment (AppleBetas) [inline Source/Managers/Modules.js:1]: sexxxx asf

File: Source/Modules/Steamboat/steamboat.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
+const config = require('./config');
```
Comment (AppleBetas) [inline Source/Modules/Steamboat/steamboat.js:1]: WHY is THIS called STEAMBOAT.
STEAMBOAT is JUST another BOT

File: Source/Util/Command.js [hunk 4]
```diff
@@ -16,9 +16,9 @@ class Command {
     }
 
     userCanAccess(member) {
-        let isAdmin = member.client.config.admins.includes(member.id);
+        var isAdmin = member.client.config.admins.includes(member.id);
         if(this.requiresAdmin && !isAdmin) return false;
-        if(member) return (member.hasPermissions(this.requiredPermissions)) || isAdmin;
+        if(member) return (member.hasPermission(this.requiredPermissions)) || isAdmin;
```
Comment (AppleBetas) [inline Source/Util/Command.js:21]: Wait, what?
`requiredPermissions` is an array, why did you change this to `hasPermission`?

Comment (EricRabil) [reply Source/Util/Command.js:21]: It was changed in the latest version of discord.JS, hasPermissions is deprecated because hasPermission accepts an array

Comment (AppleBetas) [reply Source/Util/Command.js:21]: Ah

**Review by AppleBetas — APPROVED** [review]
